### DISCUSSION TITLE
How does pyinstaller handles UI automation when making exe in windows?

### DISCUSSION THREAD

**Discussion by ReetiMauryaCrest** [pyinstaller/pyinstaller#8588] [PyInstaller] [answered]
I am currently working on an automation task using pywinauto. In this task I am required to automate a windows exe which is hard to access for me, thus I am working on making a simulator which somewhat behaves like the original exe, allowing me to test my flow, debug errors and such.

I am able to make GUI application easily using Kivy library and am packaging it using pyinstaller. But after the executable is made, I cannot get any information about controls and such using pywinauto. Thus I want to know how does pyinstaller manages automation ids and control type information while packaging an app, and if it is even possible to get automation id and control type names from an exe which is packaged with pyinstaller?

**Comment by bwoodsend** [comment, marked as answer]
It doesn't have anything to do with them. If you think PyInstaller is somehow passively interfering with them then you'll need to provide a minimal example of what you're doing so that we can debug.

**Reply by ReetiMauryaCrest** [reply]
I don't think pyinsytaller is interferring with that, I just didn't knew that how pyinstaller and kivy when used together works to make the windows application. If the pyinstaller is not working with that, then I think the kivy platform does not support making automation ids for its control. I will look into other methods to make my simulator app then. Thank you.
